### PR TITLE
add mag_pert to source_position PointSource

### PR DESCRIPTION
**Pull request by smericks**
Added ability to apply magnification perturbations to images from a PointSource of type SOURCE_POSITION. This requires passing in a point source kwarg called 'mag_pert'. This change will allow users to make images from point sources with magnification errors.

### CODE REVIEW

File: test/test_PointSource/test_Types/test_ps_base.py [hunk 1]
```diff
@@ -45,6 +45,15 @@ def test_expand_to_array(self):
...
         assert array_out[1] == 1
 
+    def test_shrink_arra(self):
+
+        from lenstronomy.PointSource.Types.base_ps import _shrink_array
+        array = [1, 2, 3]
+        num = 2
+        array_out = _shrink_array(array, num)
+        assert len(array_out) == num
+
```
Comment (sibirrer) [inline test/test_PointSource/test_Types/test_ps_base.py:55]: Can you add tests for the other scenarios of this function

File: lenstronomy/PointSource/Types/base_ps.py [hunk 2]
```diff
@@ -94,3 +94,18 @@ def _expand_to_array(array, num):
         return out
     else:
         return array
+
+
+def _shrink_array(array, num):
+    """
+
```
Comment (sibirrer) [inline lenstronomy/PointSource/Types/base_ps.py:101]: please document this definition

File: lenstronomy/PointSource/Types/source_position.py [hunk 3]
```diff
@@ -77,6 +77,9 @@ def image_amplitude(self, kwargs_ps, kwargs_lens=None, x_pos=None, y_pos=None, m
             point_amp = kwargs_ps['point_amp']
             if x_pos is not None:
                 point_amp = _expand_to_array(point_amp, len(x_pos))
+        mag_pert = kwargs_ps.get('mag_pert', 1)
```
Comment (sibirrer) [inline lenstronomy/PointSource/Types/source_position.py:80]: please add a description of this variable/option in the definition of the SourcePosition class

**Review by sibirrer — CHANGES_REQUESTED** [review]
Thank you very much! Overall looks good. I have a few requests on testing and documentation, pending the integrated tests pass.

Also: please add your name and GitHub link to the AUTHORS.rst file
Simon

**Comment by smericks** [comment]
Thank you for the comments. All of the changes you've requested should be fixed. Please let me know if there are more changes needed.

Best,
Sydney

**Review by sibirrer — APPROVED** [review]

**Comment by sibirrer** [comment]
Thank you very much @smericks and congratulations for your first contributions to lenstronomy!

**Pull request merged**